oauth-common/src/main/java/io/strimzi/kafka/oauth/services/ValidatorKey.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2017-2020, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.services;
+
+import java.util.Objects;
+
+public class ValidatorKey {
+
+    String validIssuerUri;
+    String usernameClaim;
+    String fallbackUsernameClaim;
+    String fallbackUsernamePrefix;
+
+    String sslTruststore;
+    String sslStorePassword;
+    String sslStoreType;
+    String sslRandom;
+    boolean hasHostnameVerifier;
+
+
+    ValidatorKey() {}
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof ValidatorKey)) return false;
+        ValidatorKey that = (ValidatorKey) o;
+        return hasHostnameVerifier == that.hasHostnameVerifier &&
+                Objects.equals(validIssuerUri, that.validIssuerUri) &&
+                Objects.equals(usernameClaim, that.usernameClaim) &&
+                Objects.equals(fallbackUsernameClaim, that.fallbackUsernameClaim) &&
+                Objects.equals(fallbackUsernamePrefix, that.fallbackUsernamePrefix) &&
+                Objects.equals(sslTruststore, that.sslTruststore) &&
+                Objects.equals(sslStorePassword, that.sslStorePassword) &&
+                Objects.equals(sslStoreType, that.sslStoreType) &&
+                Objects.equals(sslRandom, that.sslRandom);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(validIssuerUri, usernameClaim, fallbackUsernameClaim, fallbackUsernamePrefix, sslTruststore, sslStorePassword, sslStoreType, sslRandom, hasHostnameVerifier);
+    }
+
+
+    public static class JwtValidatorKey extends ValidatorKey {
+
+        String jwksEndpointUri;
+        int jwksRefreshSeconds;
+        int jwksExpirySeconds;
+        int jwksRefreshMinPauseSeconds;
+        boolean checkAccessTokenType;
+        boolean enableBouncy;
+        int bouncyPosition;
```

Comment:
I removed the builders, and made these private final.

Reply:
Just to be clear, my problem was with having builders _and_ mutable fields. If the fields are mutable then there's no need really for the builder. If the fields are immutable then using a builder to avoid a ctor with a stupid number of parameters is OK. Or you just have the constructor with the stupid number of parameters. Since the class is only ever instantiated once then having the constructor is acceptable (to me at least). One benefit of the ctor is it forces the caller to provide all the arguments. You can do that with the builder too, of course, but it becomes a runtime error and it make the builder more complicated.